fitlins/interfaces/afni.py

```diff
@@ -339,6 +332,26 @@ def get_stim_labels(self):
         weights = _flatten([x["weights"] for x in self.inputs.contrast_info])
         return list(set(_flatten([x.keys() for x in weights])))
 
+    def save_tsnr(self, runtime, rbetas, rvars):
+        vol_labels = parse_afni_ext(rbetas)["BRICK_LABS"].split("~")
+        mat = pd.read_csv(self.inputs.design_matrix, delimiter="\t", index_col=0)
+        # find the name of the constant column
+        const_name = mat.columns[(mat != 1).sum(0) == 0].values[0]
```

Comment:
Does this choke if there are non-steady-state volumes? Generally the constant should be zeroed out like everything else if you're censoring a volume.

Reply:
In addition to verifying that zeroing out everything else or not doesn't impact the estimates of the other coefficients, I also checked to see if we should expect a design matrix that doesn't have an all 1's constant column. We're getting the design matrix from nilearn's `make_first_level_design_matrix` function, which adds a an all 1s constant column as part of the drift model function, even if the drift model is None:
https://github.com/nilearn/nilearn/blob/d2ad41a9fa0aa7041c7e826518c6c3692a92a2e1/nilearn/glm/first_level/design_matrix.py#L122-L173

Based on that, I think this logic is fine.